File: CivitasJava/src/civitas/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author luis
 */
public abstract class Sorpresa {
    private String texto;
    
    Sorpresa(String texto){
        this.texto = texto;
    }
    
    public abstract void aplicarAJugador(int actual, ArrayList<Jugador> todos);
            
    void informe(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            Diario diary = Diario.getInstance();
            diary.ocurreEvento("Aplicando sorpresa a: " + todos.get(actual).getNombre());
        }   
    }
            
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        if(actual >= 0 && actual  < todos.size())
            return true;
        else
            return false;
    }
    
    @Override
    public String toString(){
        String e = "\nNombre: " + this.texto;
        return e;
    }
}

File: CivitasRuby/pkg/CivitasRuby-0.0.1/lib/sorpresa.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'jugador.rb'
require_relative 'jugador_especulador.rb'

module Civitas
  
  class Sorpresa
    
    
    def initialize(texto)
      @texto = texto
    end
    
    def informe(actual, todos)
      if jugador_correcto(actual,todos)
        diary = Diario.instance
        diary.ocurre_evento("Aplicando sorpresa #{@tipo.to_s} a: #{@nombre}")
      end
    end
    
    def jugador_correcto(actual, todos)
      if actual >= 0 && actual < todos.length
        return true
      else
        return false
      end
    end
   
    
    def to_s
      "\nNombre: #{@texto}"
    end
    
    private_class_method  :new
  end

end
